### PR TITLE
[BOOKINGSG-5584][RT] Fix date input color after selection in mobile stays gray

### PR DESCRIPTION
**Pull request by raymondtanwl**
**Changes**
- Fix bug involving date input on mobile view. The text color remain gray after user selected a date due to some hover function not fired in mobile view.
- [delete] branch

**Changelog entry**
- Fix bug involving date input on mobile view.

**Additional information**
- You may refer to this [BOOKINGSG-5884](https://jira.ship.gov.sg/browse/BOOKINGSG-5884)

### CODE REVIEW

File: src/date-input/date-input.tsx [hunk 1]
```diff
@@ -161,6 +161,10 @@ export const DateInput = ({
         if (onChange) {
             onChange(changeValue);
         }
+        // clear hover value for mobile when onMouseLeave={handleMouseLeaveCell} is not triggered due to touch input
+        if (!withButton && hoveredDate) {
+            setHoveredDate(undefined);
+        }
```
Comment (qroll) [inline src/date-input/date-input.tsx:167]: any specific reason for putting this in `performOnChangeHandler` vs `handleSelect`?

**Review by qroll — COMMENTED** [review]
`DateRangeInput` is similar, could I trouble you to apply the changes to that component as well?

File: src/date-input/date-input.tsx (hunk 1 above)
Comment (raymondtanwl) [reply src/date-input/date-input.tsx:167]: no specific reason, i can move it to `handleSelect` too

**Comment by raymondtanwl** [comment]
> `DateRangeInput` is similar, could I trouble you to apply the changes to that component as well?

@qroll unable to replicate issue for `DateRangeInput` as the calendar still show button in mobile view and clicking the button  works as expected.

File: src/date-input/date-input.tsx (hunk 1 above)
Comment (qroll) [reply src/date-input/date-input.tsx:167]: ok, would suggest to move this logic out to keep `performOnChangeHandler` single-purpose

Comment (raymondtanwl) [reply src/date-input/date-input.tsx]: done

**Review by qroll — APPROVED** [review]

**Pull request merged**